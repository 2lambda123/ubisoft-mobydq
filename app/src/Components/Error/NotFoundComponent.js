import React from 'react';
import {styles} from './../../styles/baseStyles';
import {withStyles} from '@material-ui/core/styles';

class NotFoundComponent extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <React.Fragment>
                <br/><br/><br/>
                <h1>404 Not Found</h1>
            </React.Fragment>
        )
    }
}

export default withStyles(styles)(NotFoundComponent);
