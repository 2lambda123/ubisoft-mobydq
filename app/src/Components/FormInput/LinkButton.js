import React from 'react';
import Button from '@material-ui/core/Button';

import { Link } from 'react-router-dom';

const LinkButton = ({
  type,
  color,
  variant,
  disabled,
  label,
  touched, // eslint-disable-line
  to
}) => <Button type={type} component={Link} to={to} disabled={disabled} variant={variant} color={color}>
  {label}
</Button>;

export default LinkButton;
