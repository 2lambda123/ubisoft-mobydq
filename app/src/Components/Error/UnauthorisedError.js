import React from 'react';
import './../../styles/baseStyles';
import LinkButton from './../FormInput/LinkButton';

export const UnauthorisedError = () => <div style={{ 'display': 'flex', 'justifyContent': 'center', 'alignItems': 'center' }}>
  <div>
    <h4>Login expired</h4>
    <LinkButton disabled={false} label="Click here to re-login" type="Create" color="primary"
      variant="contained" to={'/login'}/>
  </div>
</div>;
