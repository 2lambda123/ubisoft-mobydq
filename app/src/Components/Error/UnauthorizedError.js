import React from 'react';
import './../../styles/baseStyles';
import LinkButton from './../FormInput/LinkButton';

export const UnauthorizedError = () => <div style={{ 'display': 'flex', 'justifyContent': 'center', 'alignItems': 'center' }}>
  <div>
    <div style={{ 'textAlign': 'center' }}>
      Your session has expired
    </div>
    <div style={{ 'textAlign': 'center' }}>
      <LinkButton label="Re-login" type="Create" color="secondary" variant="contained" to={'/login'}/>
    </div>
  </div>
</div>;
