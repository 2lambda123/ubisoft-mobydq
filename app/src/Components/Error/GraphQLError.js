import React from 'react';
import { UnauthorisedError } from './UnauthorisedError';

class GraphQLError extends React.Component {
  render() {
    const error = this.props.error;
    if (error.graphQLErrors.length > 0) {
      return <div>
        {error.graphQLErrors.statusCode + ': ' + error.graphQLErrors.result.message}
      </div>;
    }
    if (error.networkError) {
      switch (error.networkError.statusCode) {
        case 401:
          return <UnauthorisedError/>;
        default:
          return <div>
            {error.networkError.statusCode + ': ' + error.networkError.result.message}
          </div>;
      }
    }
    return <div>No error. Something wrong happened...</div>;
  }
}

export default GraphQLError;
