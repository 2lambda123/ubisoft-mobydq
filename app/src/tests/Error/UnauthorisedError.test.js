import React from 'react';
import { shallowWrap } from './../../setupTests';

import { MemoryRouter } from 'react-router-dom';

import { UnauthorizedError } from '../../Components/Error/UnauthorizedError';
import LinkButton from './../../Components/FormInput/LinkButton';

describe('UnauthorizedError component unit test', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWrap(
      <UnauthorizedError/>
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('UnauthorizedError component functional test', () => {
  it('renders the UnauthorizedError component', () => {
    const wrapper = mount(<div><MemoryRouter><UnauthorizedError/></MemoryRouter></div>);
    const loginButton = <LinkButton label="Re-login" color="secondary" variant="contained" to={'/login'}/>;
    expect(wrapper.contains(loginButton)).toEqual(true);
  });
});