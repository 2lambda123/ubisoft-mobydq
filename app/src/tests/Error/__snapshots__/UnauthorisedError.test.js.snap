// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthorizedError component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UnauthorizedError />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Your session has expired
        </div>
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <LinkButton
            color="secondary"
            label="Re-login"
            to="/login"
            variant="contained"
          />
        </div>
      </div>,
      "style": Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Your session has expired
          </div>,
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <LinkButton
              color="secondary"
              label="Re-login"
              to="/login"
              variant="contained"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your session has expired",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Your session has expired",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LinkButton
              color="secondary"
              label="Re-login"
              to="/login"
              variant="contained"
            />,
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "secondary",
              "label": "Re-login",
              "to": "/login",
              "variant": "contained",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Your session has expired
          </div>
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <LinkButton
              color="secondary"
              label="Re-login"
              to="/login"
              variant="contained"
            />
          </div>
        </div>,
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Your session has expired
            </div>,
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <LinkButton
                color="secondary"
                label="Re-login"
                to="/login"
                variant="contained"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your session has expired",
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "Your session has expired",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <LinkButton
                color="secondary"
                label="Re-login"
                to="/login"
                variant="contained"
              />,
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "secondary",
                "label": "Re-login",
                "to": "/login",
                "variant": "contained",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
}
`;
